src/raftstore/store/peer.rs

```diff
@@ -1043,11 +1115,37 @@ impl Peer {
                uuid,
                index);
 
-        if cb.is_none() {
+        if cmd_cb.is_none() {
             return exec_result;
         }
 
-        let cb = cb.unwrap();
+        let (cb, lease_renew_time) = cmd_cb.unwrap();
+        // Try to renew the leader lease as this command asks to.
+        if let Some(current_expired_time) = self.leader_lease_expired_time {
+            // This peer is leader and has recorded leader lease.
+            // Calculate the renewed lease for this command. If the renewed lease lives longer
+            // than current leader lease, update the leader lease to the renewed lease.
+            let next_expired_time = self.next_lease_expired_time(lease_renew_time);
+            // Use the lease expired timestamp comparation here, so that these codes still
```

Comment:
Use the lease expired timestamp comparation here -> Use the lease expiring timestamp comparison here